settings.php

```diff
@@ -407,6 +407,19 @@
  */
 $settings['backdrop_drupal_compatibility'] = TRUE;
 
+/**
+ * Warn if multiple versions of the same module are present.
+ *
+ * If multiple versions of the same module are present (for example, a version
+ * of a module in the /modules directory that has the same name as a /core
+ * module), Backdrop will load only the last module encountered when Backdrop
+ * scans for module files and Backdrop will show a warning on the status report
+ * page. For some multisite configurations, one might wish to intentionally
+ * override a core or contrib module with a different version in the /sites
+ * folder. To disable the status report warning, set this value to FALSE.
+ */
+$settings['multiple_module_version_warnings'] = TRUE;
```

Comment:
A few notes/suggestions:
- The first sentence is really long, and without punctuation it becomes difficult to read.
- We could split this huge paragraph into two smaller ones, one to explain how module discovery works and when the warnings are thrown, and another to explain the situations when it might be useful to disable the warnings.
- I think that it might be better to have the setting ready for silencing the warnings, but commented out.
- If the setting name was using a verb, then it'd be easier to scan and figure out what it does, and what TRUE/FALSE would do. Something like `disable_multiple_modules_warnings` perhaps.

How about something like this instead? (feel free to tweak):

```suggestion
/**
 * Suppress warnings of multiple instances of the same module being found.
 *
 * When scanning for module files, if Backdrop encounters multiple instances of
 * the same module (for example, a version of a module in the /modules directory
 * that has the same name as a module in /core), then only the last module will
 * be loaded. In such cases, Backdrop will show a warning on the status report
 * page.
 *
 * Having multiple versions of the same module may be intentional in certain use
 * cases though, such as in some multisite configurations, when there is need to
 * override a core or contrib module with a different version in the /sites
 * folder. In such cases, you may want to disable the status report warnings.
 *
 * Uncomment the line below to disable the warnings.
 */
// $settings['disable_multiple_modules_warnings'] = TRUE;
```

Reply:
I like the description change!👍

That variable change makes the call at point of use

```
if (!settings_get('disable_multiple_modules_warnings')) {...}
```

I don't like the double-negative ("if not disable..."). There's a use case for `$settings['...'] = TRUE;`, and "To disable the status report warning, set this value to FALSE" seems pretty clear, too, and goes along with the general sense of "TRUE = enable, FALSE = disable."

Reply:
After a good night's sleep, I've come around to this point of view (in part, because it makes the warning the default even for people who don't update their settings.php file). PR updated accordingly.